src/components/refactor/dialogs/line-creation/line-creation-dialog.js

```diff
@@ -0,0 +1,286 @@
+                },
+                [CURRENT_LIMITS_2]: {
+                    [PERMANENT_LIMIT]: line.currentLimits2?.permanentLimit,
+                },
+                ...getConnectivityFormData(
+                    {
+                        voltageLevelId: line.voltageLevelId1,
+                        busbarSectionId: line.busOrBusbarSectionId1,
+                        connectionDirection: line.connectionDirection1,
+                        connectionName: line.connectionName1,
+                        connectionPosition: line.connectionPosition1,
+                    },
+                    CONNECTIVITY_1
+                ),
+                ...getConnectivityFormData(
+                    {
+                        voltageLevelId: line.voltageLevelId2,
+                        busbarSectionId: line.busOrBusbarSectionId2,
+                        connectionDirection: line.connectionDirection2,
+                        connectionName: line.connectionName2,
+                        connectionPosition: line.connectionPosition2,
+                    },
+                    CONNECTIVITY_2
+                ),
+            });
+        },
+        [reset]
+    );
+
+    const searchCopy = useFormSearchCopy({
+        studyUuid,
+        currentNodeUuid,
+        equipmentPath,
+        toFormValues: (data) => data,
+        setFormValues: fromSearchCopyToFormValues,
+    });
+
+    useEffect(() => {
+        if (editData) {
+            fromEditDataToFormValues(editData);
+        }
+    }, [fromEditDataToFormValues, editData]);
+
+    const onSubmit = useCallback(
+        (line) => {
+            createLine(
+                studyUuid,
+                currentNodeUuid,
+                line[EQUIPMENT_ID],
+                sanitizeString(line[EQUIPMENT_NAME]),
+                line.seriesResistance,
+                line.seriesReactance,
+                line.ShuntConductance1,
+                line.shuntSusceptance1,
+                line.ShuntConductance2,
+                line.shuntSusceptance2,
+                line.connectivity1?.voltageLevel?.id,
+                line.connectivity1?.busOrBusbarSection?.id,
+                line.connectivity2?.voltageLevel?.id,
+                line.connectivity2?.busOrBusbarSection?.id,
+                line.currentLimits1?.permanentLimit,
+                line.currentLimits2?.permanentLimit,
+                editData ? true : false,
+                editData ? editData.uuid : undefined,
+                line.connectivity1?.connectionName ?? null,
+                line.connectivity1?.connectionDirection ??
+                    UNDEFINED_CONNECTION_DIRECTION,
+                line.connectivity2?.connectionName ?? null,
+                line.connectivity2?.connectionDirection ??
+                    UNDEFINED_CONNECTION_DIRECTION,
+                line.connectivity1?.connectionPosition ?? null,
+                line.connectivity2?.connectionPosition ?? null
```

Comment:
use field constants to make field renaming easier in the future

Reply:
done